databend-jdbc/src/main/java/com/databend/jdbc/DatabendResultSet.java

```diff
@@ -151,6 +151,11 @@ public AsyncIterator(Iterator<T> dataIterator, DatabendClient client)
                     finished = true;
                 }
             });
+            for(;;){
```

Comment:
It seems that potentially causing the application to block during startup, because the test CI was blocked. It is generally not recommended to perform such waiting operations in a constructor because they may block the main thread. And I tested running more than one sql in a statemnet, I can't reproduce it.

Reply:
I will separate this branch to avoid conflict